### ISSUE TITLE
Pandas and NumPy compatibility issues

### ISSUE THREAD

**Issue by dqmota** [totvslabs/pyCarol#610]
Dependencies should be set to the following versions to work.

- pandas==1.5.3
- numpy==1.23.5

```
File [.venv/lib/python3.11/site-packages/pycarol/staging.py:152](https://file+.vscode-resource.vscode-cdn.net/.venv/lib/python3.11/site-packages/pycarol/staging.py:152), in Staging.send_data(self, staging_name, data, connector_name, connector_id, step_size, print_stats, gzip, auto_create_schema, crosswalk_auto_create, flexible_schema, force, max_workers, dm_to_delete, async_send, carol_data_storage, storage_only, carol_sync)
    150 else:
    151     overwrite = True
--> 152 self.create_schema(_sample_json, staging_name, connector_id=connector_id, export_data=carol_data_storage,
    153                    crosswalk_list=crosswalk_auto_create, overwrite=overwrite, mdm_flexible=flexible_schema)
    154 _crosswalk = crosswalk_auto_create
    155 print('provided crosswalk ', _crosswalk)

File [.venv/lib/python3.11/site-packages/pycarol/staging.py:293](https://file+.vscode-resource.vscode-cdn.net/.venv/lib/python3.11/site-packages/pycarol/staging.py:293), in Staging.create_schema(self, fields_dict, staging_name, connector_id, connector_name, mdm_flexible, crosswalk_name, crosswalk_list, overwrite, auto_send, export_data, data)
    291 elif isinstance(data, str):
    292     schema = carolSchemaGenerator.from_json(data)
--> 293     schema = schema.to_dict(mdmStagingType=staging_name, mdmFlexible=mdm_flexible,
    294                             crosswalkname=crosswalk_name, crosswalkList=crosswalk_list)
    295 else:
    296     print('Behavior for type %s not defined!' % type(data))

File [.venv/lib/python3.11/site-packages/pycarol/schema_generator.py:118](https://file+.vscode-resource.vscode-cdn.net/.venv/lib/python3.11/site-packages/pycarol/schema_generator.py:118), in carolSchemaGenerator.to_dict(self, mdmStagingType, mdmFlexible, crosswalkname, crosswalkList, export_data)
    115 schema_dict.update({"mdmExportData": export_data})
    117 for prop, value in base_object.items():
--> 118     schema_dict["mdmStagingMapping"]["properties"].update({prop: _dictConstructor(base_object=value)})
    120 self.schema_dict = schema_dict
    121 return schema_dict

File [.venv/lib/python3.11/site-packages/pycarol/schema_generator.py:137](https://file+.vscode-resource.vscode-cdn.net/.venv/lib/python3.11/site-packages/pycarol/schema_generator.py:137), in _dictConstructor(base_object)
    135 schema_dict = {}
    136 base_object_type = type(base_object)
--> 137 schema_type = Type.get_schema_type_for(base_object_type)
    139 # schema_dict["required"] = True
    140 schema_dict["type"] = schema_type.json_type

File [.venv/lib/python3.11/site-packages/pycarol/schema_generator.py:55](https://file+.vscode-resource.vscode-cdn.net/.venv/lib/python3.11/site-packages/pycarol/schema_generator.py:55), in Type.get_schema_type_for(cls, t)
     41 import numpy as np
     43 """docstring for get_schema_type_for"""
     45 SCHEMA_TYPES = {
     46     type(None): NullType,
     47     str: StringType,
     48     int: IntType,
     49     float: DoubleType,
     50     bool: BooleanType,
     51     list: ArrayType,
     52     dict: ObjectType,
     53     np.float64: DoubleType,
     54     np.float32: DoubleType,
---> 55     np.float: DoubleType,
     56     np.int64: IntType,
     57     np.int32: IntType,
     58     np.int: IntType,
     59     np.bool_: BooleanType,
     60     datetime: DateType,
     61     pd.Timestamp:DateType
     62 }
     64 schema_type = SCHEMA_TYPES.get(t)
     66 if not schema_type:

File [.venv/lib/python3.11/site-packages/numpy/__init__.py:313](https://file+.vscode-resource.vscode-cdn.net/.venv/lib/python3.11/site-packages/numpy/__init__.py:313), in __getattr__(attr)
    308     warnings.warn(
    309         f"In the future `np.{attr}` will be defined as the "
    310         "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
    312 if attr in __former_attrs__:
--> 313     raise AttributeError(__former_attrs__[attr])
    315 if attr == 'testing':
    316     import numpy.testing as testing

AttributeError: module 'numpy' has no attribute 'float'.
`np.float` was a deprecated alias for the builtin `float`. To avoid this error in existing code, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
```

**Comment by RWallan** [comment]
This problem has also mentioned in this issue #607 . And a pull request is opened with a possible solution to fix that: #609. 

I'm available to discuss the solution suggested in this PR to make the best solution.

**Comment by darolt** [comment]
fixed on pycarol 2.55.0.

**Comment by darolt** [comment]
fixed on pycarol 2.55.0.
